File: October_4/Solution.py
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root==None:
            return False
                                    
        targetSum=targetSum-root.val
                                                    
        if root.left==None and root.right==None:
            return targetSum==0
                                                                                
        return self.hasPathSum(root.left,targetSum) or self.hasPathSum(root.right,targetSum)

File: October_4/Solution.cs
public class Solution {
    public bool HasPathSum(TreeNode root, int targetSum) {
        return dfs(root, targetSum);
    }
    public bool dfs(TreeNode root, int target){
        if(root == null){
            return false;
        }
        int newTarget = target-root.val;
        if(newTarget == 0 && root.left == null && root.right ==null){
            return true;
        }
        return dfs(root.left, newTarget) || dfs(root.right, newTarget);
    }
}
